velox/exec/ExchangeClient.cpp

```diff
@@ -48,6 +48,9 @@ void ExchangeClient::addRemoteTaskId(const std::string& taskId) {
     } else {
       sources_.push_back(source);
       queue_->addSourceLocked();
+      // Put new source into 'producingSources_' queue to prioritise fetching
+      // from these to find out whether these are productive or not.
+      producingSources_.push(source);
```

Comment:
Should we say something "Initially all sources are 'producing'"?

Reply:
We can change this comment, but I'm not sure if "Initially all sources are 'producing'" provides any additional information that's not already available from the code itself.